torch/_inductor/lowering.py

```diff
@@ -734,7 +734,7 @@ def expand(x, sizes):
         return x
 
     x_size_product = V.graph.sizevars.size_hint(sympy_product(x.get_size()))
-    if x_size_product > 0:
+    if x_size_product > 0 and not any([V.graph.sizevars.shape_env.is_unbacked_symint(s) for s in sizes]):
```

Comment:
The heuristic here is confusing to me. If we think the input hints to be non-empty, why would we SUPPRESS realizing the input if some of them are unbacked? It seems to me we want to realize it.